liquidctl/driver/kraken3.py

```diff
@@ -788,8 +789,15 @@ def check_unsupported_fw_version():
                 self._get_fw_version()
                 if self.fw[0] == 2:
                     raise NotSupportedByDriver(
-                        "setting images is not supported on firmware 2.X.Y, please see issue #631"
+                        "gif images is not supported on firmware 2.X.Y, please see issue #631"
                     )
+        
+        def is_fw_version2():
```

Comment:
This does the same thing as check_unsupported_version(), minus the logging. Since you're now using that in only one place, I'd suggest to replace that call with this and log + return if necessary.

Also the name is ambigious, I'd call it `_is_elite_fw_version2()` or similar (but prefixed with an underscore, since it's private).

Reply:
This is to determine how to send the image data, so throwing an exception would exclude firmware 1.X.Y, unless caught

Reply:
I was referring to [this part](https://github.com/liquidctl/liquidctl/blob/3aba011146eb0432b9db786ecd2fcc94e5f16fc8/liquidctl/driver/kraken3.py#L833). Now that you have `_is_2023_fw_version2()`, use it there instead of `check_unsupported_fw_version()` and raise the same exception. That way, the [fw checking code ](https://github.com/liquidctl/liquidctl/blob/3aba011146eb0432b9db786ecd2fcc94e5f16fc8/liquidctl/driver/kraken3.py#L784-L800) is not duplicated.

Reply:
I see